src/sql/workbench/contrib/notebook/browser/notebookLinkHandler.ts

```diff
@@ -173,9 +174,17 @@ export class NotebookLinkHandler {
  */
 export function findPathRelativeToContent(notebookFolder: string, contentPath: URI | undefined, isMarkdown?: boolean, isEncoded?: boolean): string {
 	if (contentPath?.scheme === 'file') {
-		let relativePath = contentPath.fragment ? path.relative(notebookFolder, contentPath.fsPath).concat('#', contentPath.fragment) : path.relative(notebookFolder, contentPath.fsPath);
+		let decodePath: boolean = contentPath.fsPath !== decodeURI(contentPath.fsPath);
```

Comment:
Some comments here explaining why this is necessary would be helpful...especially since I'm confused with how `isEncoded` fits into the picture. Why is that not being used to tell us whether we need to decode the path?

Why are we decoding the fsPath anyways? That value generally should already be decoded - for example : 

`const testUri = vscode.Uri.file('C:/Test Path With Space/file.txt');`

results in these : 

![image](https://user-images.githubusercontent.com/28519865/214901931-4f9a0080-0545-47de-ad17-731fce109cf4.png)

Or are we doing something weird with pre-encoding the path we create the URI out of which requires it be decoded first? And are you making sure that this isn't incorrectly decoding paths that have % characters in them?

Reply:
the isEncoded is set by reading the is-encoded attribute value of the link (am not really sure when that'll be true). The contentParth passed here is nothing but the _notebookUriLink which is explicitly encoded [here](https://github.com/microsoft/azuredatastudio/blob/0e65f0d7eb555d6d9ae8eb5504965b7f0f65afd1/src/sql/workbench/contrib/notebook/browser/notebookLinkHandler.ts#L63), Which is why I needed to decode it here.

Reply:
Hey Maddy, I found this image to help understand more of the Notebook linking flow:
<img width="892" alt="image" src="https://user-images.githubusercontent.com/23587151/215204093-b10c15df-b053-4947-9803-a00185fcde4b.png">

This was made a while back but hopefully can help further understand the linking process of the Notebook flow and how it was used to fix many issues.